frontend/src/metabase/visualizations/components/Visualization/Visualization.jsx

```diff
@@ -216,13 +216,17 @@ class Visualization extends PureComponent {
       : undefined;
   }
 
-  getMode(maybeModeOrQueryMode, question) {
-    if (maybeModeOrQueryMode instanceof Mode) {
-      return maybeModeOrQueryMode;
+  getMode(modeOrModeGetter, question) {
+    if (typeof modeOrModeGetter === "function") {
+      modeOrModeGetter = modeOrModeGetter({ question });
```

Comment:
We need a getter here to properly use `Question` when creating the `new Mode` inside the `frontend/src/metabase/visualizations/click-actions/lib/modes.ts`